tests/rest/client/test_typing.py

```diff
@@ -43,30 +48,34 @@ def make_homeserver(self, reactor, clock):
 
         self.event_source = hs.get_event_sources().sources.typing
 
-        hs.get_federation_handler = Mock()
+        hs.get_federation_handler = Mock()  # type: ignore[assignment]
 
-        async def get_user_by_access_token(token=None, allow_guest=False):
-            return {
-                "user": UserID.from_string(self.auth_user_id),
-                "token_id": 1,
-                "is_guest": False,
-            }
+        async def get_user_by_access_token(
+            token: str,
+            rights: str = "access",
+            allow_expired: bool = False,
+        ) -> TokenLookupResult:
+            return TokenLookupResult(
+                user_id=self.user_id,
+                is_guest=False,
+                token_id=1,
+            )
```

Comment:
Do I understand correctly---it looks we were were patching ina mock which had the wrong function signature? If so, why didn't the old test fail?

Reply:
Yes, the function had the wrong signature.

IMO `TokenLookupResult` is also an `dict` with the same content.
https://github.com/matrix-org/synapse/blob/e24ff8ebe3d4119d377355402245947f7de61c00/synapse/storage/databases/main/registration.py#L55-L79